### ISSUE TITLE
Use invidious apis dynamically 

### ISSUE THREAD

**Issue by notnotrachit** [notnotrachit/youlearn#3]
Currently we are using one particular invidious server api, however, we should use the invidious API to get the best server & use that everytime

**Comment by lodenrogue** [comment]
I'm looking through the API response here: https://api.invidious.io/instances.json?pretty=1&sort_by=api,health

What determines whether a server is "best"?

**Comment by notnotrachit** [comment]
Yup, thats exactly I was saying, the most active api with the highest health
